src/components/profiles-modals/DeleteProfileModal.vue

```diff
@@ -0,0 +1,60 @@
+<script lang="ts">
+import { Vue, Component } from 'vue-property-decorator';
+import { ModalCard } from "../all";
+import R2Error from "../../model/errors/R2Error";
+import FsProvider from "../../providers/generic/file/FsProvider";
+
+let fs: FsProvider;
+
+@Component({
+    components: {ModalCard}
+})
+export default class DeleteProfileModal extends Vue {
+    async created() {
+        fs = FsProvider.instance;
+    }
+
+    get isOpen(): boolean {
+        return this.$store.state.modals.isDeleteProfileModalOpen;
+    }
+
+    get profileList(): string[] {
+        return this.$store.state.profiles.profileList;
+    }
+
+    closeDeleteProfileModal() {
+        this.$store.commit('closeDeleteProfileModal');
+    }
+
+    async removeProfile() {
+        try {
+            await this.$store.dispatch('profiles/removeSelectedProfile', {fs: fs, store: this.$store});
```

Comment:
- Is there any specific reason for passing the `fs` as a parameter? AFAIK the ProfilesModule can access the FsProvider directly internally. I think it would be clearer separation of concerns if the modal component didn't know about the existance of the file system operations at all. We want this component to be all about the UI
- Also it seems unnecessary to pass a reference to the store to the method of the store. I'll comment more about this on the ProfilesModule file